functions/scheduled.js

```diff
@@ -0,0 +1,64 @@
+const admin = require("firebase-admin");
+
+const functions = require("firebase-functions");
+const { HttpsError } = require("firebase-functions/lib/providers/https");
+
+const { chunk, sampleSize } = require("lodash");
+
+const DEFAULT_RECENT_USERS_IN_VENUE_CHUNK_SIZE = 6;
+
+exports.aggregateUsersLocationsInVenue = functions.pubsub
+  .schedule("every 5 minutes")
+  .onRun(async () => {
+    const venuesPromise = admin
+      .firestore()
+      .collection("venues")
+      .get()
+      .then((snapshot) =>
+        snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }))
+      );
+
+    const usersPromise = admin
+      .firestore()
+      .collection("users")
+      .get()
+      .then((snapshot) =>
+        snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }))
+      );
+
+    const [users, venues] = await Promise.all([usersPromise, venuesPromise]);
+
+    // NOTE: Chunk users into batch commit to fasten the operation
+    chunk(venues, 250).forEach((venuesChunk) => {
+      const batch = admin.firestore().batch();
+
+      venuesChunk.forEach((venue) => {
+        const recentVenueUsers = users.filter(
+          (user) =>
+            user.lastVenueIdSeenIn && user.lastVenueIdSeenIn.includes(venue.id)
+        );
+
+        const recentVenueUsersCount = recentVenueUsers.length;
+
+        const venueRef = admin.firestore().collection("venues").doc(venue.id);
+
+        batch.update(venueRef, {
+          recentUserCount: recentVenueUsersCount,
+          recentUsersSample: sampleSize(
+            recentVenueUsers,
+            venue.recentUsersSampleSize ??
+              DEFAULT_RECENT_USERS_IN_VENUE_CHUNK_SIZE
+          ),
+        });
+      });
```

Comment:
Can replace `forEach` with `for( const venue of venuesChunk )`

Reply:
Replaced in https://github.com/sparkletown/sparkle/pull/2286/commits/a772e43fe3c8a00e479bbe9244fd8d5041e2314f